### ISSUE TITLE
`undefined method 'each_node' for nil` with Ruby 3.3

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop/rubocop-rspec#1774]
Project: https://github.com/AlexWayfer/static_files_toys (`main` branch)

```
> bundle exec rubocop --debug

For /home/alex/Projects/ruby/static_files_toys: configuration from /home/alex/Projects/ruby/static_files_toys/.rubocop.yml
configuration from /home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-performance-1.20.1/config/default.yml
configuration from /home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-performance-1.20.1/config/default.yml
Default configuration from /home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/config/default.yml
configuration from /home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-capybara-2.20.0/config/default.yml
undefined method `each_node' for nil
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:158:in `required_ruby_version'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:195:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:195:in `first'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:195:in `version_from_gemspec_file'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:175:in `find_version'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:29:in `initialize'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `new'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `block in source'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `detect'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:257:in `source'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/target_ruby.rb:261:in `version'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_validator.rb:64:in `target_ruby_version'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/forwardable.rb:240:in `target_ruby_version'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_obsoletion/parameter_rule.rb:36:in `applies_to_current_ruby_version?'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_obsoletion/parameter_rule.rb:22:in `violated?'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_obsoletion.rb:100:in `block in obsoletions'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_obsoletion.rb:99:in `map'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_obsoletion.rb:99:in `obsoletions'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_obsoletion.rb:42:in `reject_obsolete!'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_validator.rb:78:in `check_obsoletions'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_validator.rb:44:in `validate'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config.rb:55:in `check'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config.rb:26:in `create'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader.rb:65:in `load_file'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader.rb:158:in `inject_defaults!'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-capybara-2.20.0/lib/rubocop-capybara.rb:15:in `<top (required)>'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-rspec-2.26.1/lib/rubocop-rspec.rb:7:in `<top (required)>'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:35:in `load'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:21:in `load'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:14:in `block (2 levels) in resolve_requires'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:13:in `each'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:13:in `block in resolve_requires'
<internal:kernel>:90:in `tap'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:12:in `resolve_requires'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader.rb:53:in `load_file'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_loader.rb:112:in `configuration_from_file'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_store.rb:68:in `for_dir'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/config_store.rb:47:in `for_pwd'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:145:in `parallel_by_default!'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:49:in `block in run'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:43:in `run'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/exe/rubocop:19:in `block in <top (required)>'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/benchmark.rb:313:in `realtime'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.59.0/exe/rubocop:19:in `<top (required)>'
/home/alex/.rbenv/versions/3.3.0/bin/rubocop:25:in `load'
/home/alex/.rbenv/versions/3.3.0/bin/rubocop:25:in `<top (required)>'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/cli/exec.rb:58:in `load'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/cli/exec.rb:58:in `kernel_load'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/cli/exec.rb:23:in `run'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/cli.rb:451:in `exec'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/cli.rb:34:in `dispatch'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/cli.rb:28:in `start'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.5.3/exe/bundle:28:in `block in <top (required)>'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/home/alex/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.5.3/exe/bundle:20:in `<top (required)>'
/home/alex/.rbenv/versions/3.3.0/bin/bundle:25:in `load'
/home/alex/.rbenv/versions/3.3.0/bin/bundle:25:in `<main>'
Finished in 0.2324854860016785 seconds
```

**Comment by AlexWayfer** [comment]
I can't even execute `rubocop -V`, here is a part of `bundle list`, so versions are the latest:

```
...
  * rubocop (1.59.0)
  * rubocop-ast (1.30.0)
  * rubocop-capybara (2.20.0)
  * rubocop-factory_bot (2.25.0)
  * rubocop-performance (1.20.1)
  * rubocop-rspec (2.26.1)
...
```

And without `rubocop-rspec` in RuboCop's config everything is OK.

**Comment by pirj** [comment]
Is this only with 2.26.x?

**Comment by AlexWayfer** [comment]
> Is this only with 2.26.x?

Let me check… OK, I've found the problem.

No, it's for 2.25.x, 2.24.x, even RuboCop 1.58.x and 1.57.x.

I've changed the `TargetRubyVersion` — nothing.

But it works with Ruby 3.2.2. And this error only for Ruby 3.3.0.

**Comment by bquorning** [comment]
Ok, I was able to reproduce the issue this way:
```
docker run --rm ruby:3.3.0 /bin/bash -c "git clone https://github.com/AlexWayfer/static_files_toys.git; cd static_files_toys; bundle install; bundle exec rubocop"
```

The problem happens inside `RuboCop::TargetRuby::GemspecFile#version_from_gemspec_file`, and I found that it works fine if I remove the single heredoc from the gemspec:
```rb
	spec.description   = <<~DESC
		Toys template for static (public) files of a web-application.
	DESC
```

I’ll dig a bit deeper and see if this is an issue with either the `rubocop-ast` or the `parser` gem.

**Comment by AlexWayfer** [comment]
@bquorning OK, thanks. I'd like to dig and help by myself, but it's pretty difficult area for me, and unexpected combination of factors.

**Comment by AlexWayfer** [comment]
Fixed by parser v3.3.0.3. Thank you all.
